integration-tests/slack/post-message.js

```diff
@@ -0,0 +1,48 @@
+const fetch = require('node-fetch')
+
+
+const defaultToken = process.env.SLACK_BOT_TOKEN
+const defaultChannel = process.env.SLACK_CHANNEL_ID
+const apiUrl = 'https://slack.com/api'
+
+const postMessage = async ({ channel = defaultChannel, token = defaultToken, blocks }) => {
+  if (!channel) {
+    console.error(`**  ERROR: Missing Slack channel. Failed to post message to Slack.`)
+    return
```

Comment:
I thought about it before. If throws error here and all tests passed, job would be marked as fail. I didn't want to see that. What do you think?

Reply:
It feels like having a false alarm (due to a test failure because of a Slack notification issue) is better than risking not getting a notification when there's an actual test failure (because of a Slack notification issue that's swallowed), no?

Reply:
There's currently some mixed messaging in this function. If no channel or token is configured, it logs "ERROR: ..." but then returns. I think either of these options would be preferable:
- Missing channel / token configuration is treated as misconfiguration and errors.
- Missing channel / token configuration is treated as disabling Slack notifications and does not log an error (maybe an info or warning).

> f throws error here and all tests passed, job would be marked as fail. I didn't want to see that.

On this subject, this function currently throws an error if the Slack API request fails. Should that fail the test?

Reply:
If all tests passed but the Slack request failed. it would cause CircleCI job to fail. 
I agree with your reasoning, I'm changing this to throw error instead return.
Thank you all.